cmd/definitions/tmpl/service.tmpl

```diff
@@ -303,7 +303,13 @@ func (s *{{ $pn }})parsePair{{ $pn }}{{ $fnk }}(opts []Pair) (pair{{ $pn }}{{ $f
             err = s.formatError("{{$fn.Name}}", err {{ $path }} )
         }()
 
-        {{- template "mode_check" makeSlice $fn.ObjectMode $fn.Params.ObjectParamName }}
+        {{- if ne $fn.ObjectMode ""}}
+            {{- $modeP := $fn.ObjectMode | toPascal }}
+            if !{{ $fn.Params.ObjectParamName }}.Mode.Is{{ $modeP }}() {
+                err = services.ObjectModeInvalidError{Expected: Mode{{ $modeP }}, Actual: {{$fn.Params.ObjectParamName}}.Mode}
```

Comment:
Why we need `{{$fn.Params.ObjectParamName}}.Mode` here, can we use `o.Mode`?

Reply:
The ObjectParamName is for getting the param name of Object, since we can ensure the var of param Object is always o, maybe I should remove this func?

Reply:
Yes, remove this func helps improve the readability.